src/game_entities/shop.py

```diff
@@ -50,16 +51,18 @@ def __init__(
         name: str,
         position: Position,
         sprite_link: str,
+        shop_balance: int,
         stock: list[dict[str, any]],
         interaction: Optional[dict[str, any]] = None,
         sprite: Optional[pygame.Surface] = None,
     ) -> None:
         super().__init__(name, position, sprite_link, interaction, sprite)
+        self.shop_balance = 500
```

Comment:
Since the default amount for the shop balance is already provided as an argument in this constructor (line 54), you can just directly replace this raw value by the argument itself (`shop_balance`).